tests/Cms/System/SystemTest.php

```diff
@@ -619,11 +620,10 @@ public function testPlugins()
 	}
 
 	/**
-	 * @covers ::server
 	 * @covers ::serverSoftware
 	 * @dataProvider providerForServerSoftware
 	 */
-	public function testServer($software, $expected)
+	public function testServerSoftware($software, $expected)
```

Comment:
As the new code just passes through the environment variable 1:1, there is IMO no need for the data provider anymore. It's enough to test this without any value and with a single valid value.